#! /usr/bin/env python

PACKAGE = 'rise_6dof'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
# using strings because there seems to be no vector support (currently)
gen.add('k', str_t, 0, '1/s <=> (m/s)/m or (rad/s)/rad', '3,3,5,1.5,3,4')
gen.add('ks', str_t, 0, 'N/(m/s) or (N*m)/(rad/s)', '70,50,125,15,40,20')
gen.add('alpha', str_t, 0, '1/s', '0.3,0.1,0.3,0.01,0.01,0.05')
gen.add('beta', str_t, 0, 'N/s or (N*m)/s', '1.6,0.5,0.45,1,1.5,1')
gen.add('use_rise', bool_t, 0, '', True)
gen.add('accel_feedforward', str_t, 0, 'kg or kg*m^2', '50,50,50,0,0,0')
gen.add('vel_feedforward', str_t, 0, 'N/(m/s) or (N*m)/(rad/s)', '0,0,0,0,0,0')
gen.add('two_d_mode', bool_t, 0, '', False)

gen.add('use_radp', bool_t, 0, '', False)
gen.add('eta_c1', str_t, 0, '', '0.25')
gen.add('eta_c2', str_t, 0, '', '0.5')
gen.add('eta_a1', str_t, 0, '', '1')
gen.add('nu', str_t, 0, '', '0.25')
gen.add('adp_beta', str_t, 0, '', '0.05')
gen.add('gamma', str_t, 0, '', '800')
gen.add('zeta_dist', str_t, 0, '', '3')


exit(gen.generate(PACKAGE, 'rise_6dof', 'controller'))
